Get CI working

bowlofeggs opened the pull request:
This pull request has three commits. The first commit reworks the new CI run_tests.sh script so that it does everything needed for the tests to pass. It also enforces diff-coverage now. The next two commits fix test failures that happen in CentOS CI that don't happen in the Vagrant dev environment.

Signed-off-by: Randy Barlow <randy@electronsweatshop.com>

jeremycline commented:
Hmm, still looks like it's failing because (among other things) it's missing alembic

bowlofeggs commented:
@jeremycline Yeah I'm doing the grind of repeatedly running the tests and fixing my commit and force pushing, over and over. Also, I'm pretty sure that only my changes to run_tests are being reflected in the test run and not my changes to the yaml file. I think (for good reason) Jenkies always uses the yaml file from the develop branch. So no idea what effect those have (and I still don't know why the exit code isn't causing Jenkins to think the tests failed.)

bowlofeggs commented:
I finally have CI passing! I need to go for the day, but for the record I do plan to break this commit up into more atomic commits before asking for review.

bowlofeggs committed:
Fix run_tests.sh so that it will run the tests correctly.

This commit gets run_tests.sh to the point where it will run the
tests in our CI system correctly. It now installs all needed
dependencies, exits correctly on failure, patches debugtoolbar out
of the config files, installs Bodhi in develop mode, and runs
diff-cover to enforce 100% coverage on new code.

Signed-off-by: Randy Barlow <randy@electronsweatshop.com>

bowlofeggs committed:
Use datetime.utcnow() in tests instead of now().

The tests in bodhi.server.scripts.test_approve_testing were using
datetime.now() to set the date on updates, which worked fine for
systems using UTC. However, these tests fail on systems that are in
some other timezones, such as the CentOS CI systems. This commit
converts those tests to use utcnow() to be consistent with the way the
server code uses timezones.

Signed-off-by: Randy Barlow <randy@electronsweatshop.com>

bowlofeggs committed:
PEP-8 a few modules.

The upstream latest version of flake8 fails in a few places that
our Vagrant flake8 doesn't seem to mind. This commit adds some
whitespace to fix those spots so they will pass in CentOS CI.

Signed-off-by: Randy Barlow <randy@electronsweatshop.com>

bowlofeggs commented:
This pull request is now ready for review.

jeremycline commented on bodhi/tests/server/scripts/test_approve_testing.py:
```diff
@@ -38,8 +38,7 @@ def test_autokarma_update_meeting_time_requirements_gets_one_comment(self):
         update.status = models.UpdateStatus.testing
-        update.date_testing = datetime.now() - timedelta(days=7)
-        self.db.flush()
+        update.date_testing = datetime.utcnow() - timedelta(days=7)
```
✨

jeremycline reviewed (approved):
:shipit: so good

bowlofeggs commented:
Let the integration be continuous.
